src/item.cpp

```diff
@@ -3574,16 +3574,18 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
     if( parts->test( iteminfo_parts::ARMOR_RIGIDITY ) && is_rigid() ) {
         // if the item has no armor data it doesn't cover that part
         const islot_armor *armor = find_armor_data();
-        if( armor->rigid ) {
-            std::string coverage = _( "<bold>This armor is rigid</bold>" );
-            info.emplace_back( "ARMOR", coverage );
-        } else {
-            //only some parts are rigid
-            std::string coverage = _( "<bold>Rigid</bold>:" );
-            for( const armor_portion_data &entry : armor->sub_data ) {
-                if( entry.rigid ) {
-                    for( const sub_bodypart_str_id &sbp : entry.sub_coverage ) {
-                        coverage += string_format( _( ", <info>%s</info>" ), sbp->name );
+        if( armor != nullptr ) {
+            if( armor->rigid ) {
+                std::string coverage = _( "<bold>This armor is rigid</bold>" );
+                info.emplace_back( "ARMOR", coverage );
+            } else if( !!armor ) {
```

Comment:
Isn't this redunant with the above nullptr check?

Reply:
You are correct. I rewrote my code and forgot to remove that.